nemo-python/tests/test_backbox.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+import csv
+import os
+import math
+from os.path import dirname, exists, abspath, isfile
+
+from nmo_python import load_file, NemoEngine
+
+
+def stringify(value):
+    if not isinstance(value, float):
+        return str(value)
+
+    if math.floor(value) == value:
+        return str(math.floor(value))
+
+    return str(value)
+
+
+def end_to_end_test(test_path):
+    def run_test(self):
+        os.chdir(test_path)
+        for file in os.listdir("."):
+            if not isfile(file) or not file.endswith(".rls"):
+                continue
+
+            program_name = file.removesuffix(".rls")
+            program = load_file(file)
+            engine = NemoEngine(program)
+            engine.reason()
+
+            for relation in program.output_predicates():
+                results_file_name = os.path.join(
+                    program_name, f"{relation}.csv"
+                )
+                if not exists(results_file_name):
+                    continue
+
+                with open(results_file_name) as results_file:
+                    expected = list(csv.reader(results_file))
+
+                    for result in engine.result(relation):
+                        result = [stringify(v) for v in result]
+
+                        self.assertTrue(
+                            result in expected,
+                            f"error at {relation}",
+                        )
```

Comment:
Note that this only checks whether each result returned is contained in the expected results, not that all expected results are returned (but this can still be fine, since we're checking that in the `nemo-cli` integration tests).

Reply:
Yeah, except if the language binding implementation ends up eating some results (e.g. due to value conversion or whatnot). So I will add this. Thanks for the heads-up.